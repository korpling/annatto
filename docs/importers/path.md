# path (importer)

Add file nodes for all files in the imported directory.

## Configuration

###  corpus_name



