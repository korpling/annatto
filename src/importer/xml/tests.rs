use std::path::Path;

use insta::assert_snapshot;

use crate::{importer::xml::ImportXML, test_util::import_as_graphml_string};

#[test]
fn inline() {
    let actual = import_as_graphml_string(
        ImportXML::default(),
        Path::new("tests/data/import/generic_xml/inline/"),
        None,
    )
    .unwrap();

    assert_snapshot!(actual);
}

#[test]
fn standoff() {
    let ser = "text_from_attribute = { token = \"text\" }\nclosing_default = \" \"";
    let importer: Result<ImportXML, _> = toml::from_str(ser);
    assert!(
        importer.is_ok(),
        "Deserialization error: {:?}",
        importer.err()
    );
    let actual = import_as_graphml_string(
        importer.unwrap(),
        Path::new("tests/data/import/generic_xml/standoff/"),
        None,
    )
    .unwrap();

    assert_snapshot!(actual);
}
