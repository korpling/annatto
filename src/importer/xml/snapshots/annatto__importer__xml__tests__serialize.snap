---
source: src/importer/xml/tests.rs
expression: serialization.unwrap()
---
closing_default = ""

[text_from_attribute]
